### ISSUE TITLE
Memory usage insane increasing in 3.0a5

### ISSUE THREAD

**Issue by clankill3r** [processing/processing#3007]
If I have this simple sketch (sketch doesn't really matter), and I keep moving my mouse then after a few minutes the sketch will use more then 1gig!

3.0a4 doesn't have this problem.

I'm on OSX Yosemite.

```java
void setup() {
  size(1600,1000);
}

void draw() {
  background(0);
}
```

**Comment by davideoliveri** [comment]
I can confirm that it sucks memory in **3.05a** on **OSX 10.7.5**. No matter the size of the frame. 

Memory usage when moving the mouse inside the frame increases about 10/12 Mb each time the Activity Monitor refresh. When not moving the mouse it increases only about 100 Kb. 

In Processing **3.04a** instead it only leaks about 100/200 Kb each refresh, no matter the mouse movement.

**Comment by benfry** [comment]
Do you have "increase memory" checked in Preferences? I suspect you have it set to something high, in which case, Java's JIT won't bother to run the garbage collector because the available memory is nowhere near used.

**Comment by davideoliveri** [comment]
I had indeed this option checked but un-checking it didn't solve the issue.

**Comment by benfry** [comment]
Was able to reproduce with 3.0a5, but it's fixed in the current source due to the major rewrite, so it will be fixed when 3.0a6 is released.
